addons/website/tests/test_menu.py

```diff
@@ -159,13 +162,40 @@ def test_01_menu_page_m2o(self):
             },
         }), headers={"Content-Type": "application/json"})
 
-        self.assertFalse(menu.page_id, "M2o should have been unset as this is a reserved URL.")
-        self.assertEqual(menu.url, '/website/info', "Menu URL should have changed.")
-        self.assertEqual(page.url, page_url, "Page's URL shouldn't have changed.")
+        self.assertFalse(self.menu.page_id, "M2o should have been unset as this is a reserved URL.")
+        self.assertEqual(self.menu.url, '/website/info', "Menu URL should have changed.")
+        self.assertEqual(self.page.url, self.page_url, "Page's URL shouldn't have changed.")
 
         # 3. Edit the menu URL back to the page URL
-        data['url'] = page_url
+        data['url'] = self.page_url
         Menu.save(1, {'data': [data], 'to_delete': []})
-        self.assertEqual(menu.page_id, page,
+        self.assertEqual(self.menu.page_id, self.page,
                          "M2o should have been set back, as there was a page found with the new URL set on the menu.")
-        self.assertTrue(page.url == menu.url == page_url)
+        self.assertTrue(self.page.url == self.menu.url == self.page_url)
+
+    def test_02_menu_anchors(self):
+        # 1. Create a page with a menu
+        self.create_page_and_menu()
+        # 2. Edit the menu URL to an anchor
+        data = {
+            'id': self.menu.id,
+            'parent_id': self.menu.parent_id.id,
+            'name': self.menu.name,
+            'url': '#anchor',
+        }
+        self.authenticate("admin", "admin")
+        # `Menu.save(1, {'data': [data], 'to_delete': []})` would have been
+        # ideal but need a full frontend context to generate routing maps,
+        # router and registry, even MockRequest is not enough
+        self.url_open('/web/dataset/call_kw', data=json.dumps({
+            "params": {
+                'model': 'website.menu',
+                'method': 'save',
+                'args': [1, {'data': [data], 'to_delete': []}],
+                'kwargs': {},
+            },
+        }), headers={"Content-Type": "application/json", "Referer": "http://localhost:8069" + self.page_url})
```

Comment:
As it's now:
- used twice
- really weird (even if I wrote the first one myself) and probably to be improved one day
- will probably be used again
- duplicating the whole comment..
- will need to edit multiple place if we make it more elegant

Would be great to make it a callable encapsulated util in this file. Probably define a new method like `def simulate_rpc_save_menu(self, data, to_delete=None)` which will do this (it's fine to pass the Referer everytime IMO)